lib/improver/nbhood.py

```diff
@@ -39,69 +39,51 @@
+class Utilities(object):
 
     """
+    Utilities for neighbourhood processing.
+    """
 
-    # Max extent of kernel in grid cells.
-    MAX_KERNEL_CELL_RADIUS = 500
+    def __init__(self):
+        pass
 
-    def __init__(self, radii_in_km, lead_times=None, unweighted_mode=False):
+    @staticmethod
+    def cumulate_array(cube):
         """
-        Create a neighbourhood processing plugin that applies a smoothing
-        kernel to points in a cube.
+        Method to calculate the cumulative sum of an array, by first
+        cumulating vertically so that the largest values are in the upper row,
+        and then cumulating horizontally, so that the largest values are in the
+        right hand column.
 
         Parameters
         ----------
+        cube : Iris.cube.Cube
+            Cube to which the vertical and horizontal cumulative summing will
+            be applied.
 
-        radii_in_km : float or List (if defining lead times)
-            The radii in kilometres of the neighbourhood kernel to
-            apply. Rounded up to convert into integer number of grid
-            points east and north, based on the characteristic spacing
-            at the zero indices of the cube projection-x/y coords.
-        lead_times : None or List
-            List of lead times or forecast periods, at which the radii
-            within radii_in_km are defined. The lead times are expected
-            in hours.
-        unweighted_mode : boolean
-            If True, use a circle with constant weighting.
-            If False, use a circle for neighbourhood kernel with
-            weighting decreasing with radius.
+        Returns
+        -------
+        cube : Iris.cube.Cube
+            Cube to which the vertical and horizontal cumulative summing has
+            been applied.
 
         """
-        if isinstance(radii_in_km, list):
-            self.radii_in_km = map(float, radii_in_km)
-        else:
-            self.radii_in_km = float(radii_in_km)
-        self.lead_times = lead_times
-        if self.lead_times is not None:
-            if len(radii_in_km) != len(lead_times):
-                msg = ("There is a mismatch in the number of radii "
-                       "and the number of lead times. "
-                       "Unable to continue due to mismatch.")
-                raise ValueError(msg)
-        self.unweighted_mode = bool(unweighted_mode)
-
-    def __str__(self):
-        result = ('<NeighbourhoodProcessing: radii_in_km: {};' +
-                  'unweighted_mode: {}>')
-        return result.format(
-            self.radii_in_km, self.unweighted_mode)
+        yname = cube.coord(axis="y").name()
+        xname = cube.coord(axis="x").name()
+        cubelist = iris.cube.CubeList([])
+        for slice_2d in cube.slices([yname, xname]):
+            data = slice_2d.data
+            data_summed_vertically = (
+                np.flipud(np.cumsum(np.flipud(data), axis=0)))
```

Comment:
The use of `np.flipud` should be removed. This results in summing the array from ny to 0 rather than 0 to ny. I do not think this is what the issue intends and is contrary to the Fortran code. It will also make the related neighbourhood ticket more different from the Fortran code.